### PR TITLE
docs: Enable code examples in macros doc tests

### PR DESCRIPTION
**Pull request by jawoznia**

### CODE REVIEW

File: sylvia-derive/src/lib.rs [hunk 1]
```diff
@@ -43,99 +47,179 @@ pub(crate) fn crate_module() -> Path {
...
 ///         {
 ///             // Some dispatching implementation
+/// #           Ok(Response::new())
 ///         }
 ///     }
 /// }
+/// # fn main() {}
+/// ```
+///
+/// And similar `Query` and `Sudo` enum messages.
```
Comment (kulikthebird) [inline sylvia-derive/src/lib.rs:118]:
```suggestion
/// Similarly for `Query` and `Sudo` enum messages.
```

File: sylvia-derive/src/lib.rs [hunk 2]
```diff
@@ -43,99 +47,179 @@ pub(crate) fn crate_module() -> Path {
...
-/// * `module` - defines module name, where all generated messages would be encapsulated; no
-/// additional module would be created if not provided
+/// If however you want to restrict the interface to use specific custom message and custom query types,
+/// you can do so using `sv::custom(msg=..., query=...)` attribute explained below.
 ///
 /// ## Attributes
 ///
 /// Messages structures are generated basing on interface trait method attributed with
-/// `#[sv::msg(msg_type, ...)`. Msg attribute takes as its first argument type of message it is
+/// `#[sv::msg(msg_type)`. Msg attribute takes as its first argument type of message it is
```
Comment (kulikthebird) [inline sylvia-derive/src/lib.rs:156]:
```suggestion
/// `#[sv::msg(msg_type)]`. Msg attribute takes as its first argument type of message it is
```

File: sylvia-derive/src/lib.rs [hunk 3]
```diff
@@ -43,99 +47,179 @@ pub(crate) fn crate_module() -> Path {
...
 ///
-/// ## Parameters
+/// Although it's not required to define `ExecC` and `QueryC` types, it's recommended to do so to allow
+/// the users of the interface to customize the behavior to their liking.
 ///
-/// `interface` attribute takes optional parameters:
-/// * `module` - defines module name, where all generated messages would be encapsulated; no
-/// additional module would be created if not provided
+/// If however you want to restrict the interface to use specific custom message and custom query types,
+/// you can do so using `sv::custom(msg=..., query=...)` attribute explained below.
```
Comment (kulikthebird) [inline sylvia-derive/src/lib.rs:151]:
```suggestion
/// you can do so using `#[sv::custom(msg=..., query=...)]` attribute explained below.
```

File: sylvia-derive/src/lib.rs [hunk 4]
```diff
@@ -43,99 +47,179 @@ pub(crate) fn crate_module() -> Path {
...
 ///
 /// ## Attributes
 ///
 /// Messages structures are generated basing on interface trait method attributed with
-/// `#[sv::msg(msg_type, ...)`. Msg attribute takes as its first argument type of message it is
+/// `#[sv::msg(msg_type)`. Msg attribute takes as its first argument type of message it is
 /// supposed to handle:
 ///   * `exec` - this is execute message variant
 ///   * `query` - this is query message variant
+///   * `sudo` - this is sudo message variant
```
Comment (kulikthebird) [inline sylvia-derive/src/lib.rs:160]:
```suggestion
///   * `exec` - execute message variant
///   * `query` - query message variant
///   * `sudo` - sudo message variant
```

File: sylvia-derive/src/lib.rs [hunk 5]
```diff
@@ -159,72 +243,225 @@ fn interface_impl(_attr: TokenStream2, item: TokenStream2) -> TokenStream2 {
...
-/// `#[sv::msg(msg_type, ...)`. Msg attribute takes as its first argument type of message it is
+/// `#[sv::msg(msg_type)`. Msg attribute takes as its first argument type of message it is
 /// supposed to handle:
 /// * `instantiate` - this is instantiation message handler. There should be always exactly one
 /// * `exec` - this is execute message variant
 /// * `query` - this is query message variant
 /// * `migrate` - this is migrate message variant
-/// handler for this kind of message.
+/// * `reply` - this is reply message variant
+/// * `sudo` - this is sudo message variant
```
Comment (kulikthebird) [inline sylvia-derive/src/lib.rs:415]:
```suggestion
/// * `reply` - reply message variant
/// * `sudo` - sudo message variant
```

File: sylvia-derive/src/lib.rs [hunk 6]
```diff
@@ -159,72 +243,225 @@ fn interface_impl(_attr: TokenStream2, item: TokenStream2) -> TokenStream2 {
...
-/// ## Parameters
-///
-/// `contract` attribute takes optional parameters:
-/// * `module` - defines module name, where all generated messages would be encapsulated; no
-/// additional module would be created if not provided
+/// And appropriate messages for `exec`, `query`, `migrate`, `reply` and `sudo` variants.
 ///
 /// ## Attributes
 ///
 /// Messages structures are generated basing on specific implemented methods attributed with
```
Comment (kulikthebird) [inline sylvia-derive/src/lib.rs:407]:
```suggestion
/// Message structures are generated based on specific implemented methods attributed with
```

File: sylvia-derive/src/lib.rs [hunk 7]
```diff
@@ -159,72 +243,225 @@ fn interface_impl(_attr: TokenStream2, item: TokenStream2) -> TokenStream2 {
...
-/// `#[sv::msg(msg_type, ...)`. Msg attribute takes as its first argument type of message it is
+/// `#[sv::msg(msg_type)`. Msg attribute takes as its first argument type of message it is
 /// supposed to handle:
 /// * `instantiate` - this is instantiation message handler. There should be always exactly one
 /// * `exec` - this is execute message variant
 /// * `query` - this is query message variant
 /// * `migrate` - this is migrate message variant
```
Comment (kulikthebird) [inline sylvia-derive/src/lib.rs:413]:
```suggestion
/// * `instantiate` - instantiation message handler. There should be always exactly one
/// * `exec` - execute message variant
/// * `query` - query message variant
/// * `migrate` - migrate message variant
```

**Review by kulikthebird — CHANGES_REQUESTED** [review]
Few text-related suggestions. Other than that lgtm

**Review by kulikthebird — APPROVED** [review]

**Comment by jawoznia** [comment]
#211

**Pull request merged**